### PR TITLE
Always format 8-byte oids as ints in hexadecimal.

### PR DESCRIPTION
**Pull request by jamadden**
Fixes #95

Built on #97 as they touch the same code.

### CODE REVIEW

File: persistent/cPersistence.c [hunk 1]
```diff
@@ -1446,23 +1451,50 @@ Per_repr(cPersistentObject *self)
...
+        oid_bytes = (unsigned char*)PyBytes_AS_STRING(self->oid);
+        oid_value = ((unsigned long long)oid_bytes[0] << 56)
+            | ((unsigned long long)oid_bytes[1] << 48)
+            | ((unsigned long long)oid_bytes[2] << 40)
+            | ((unsigned long long)oid_bytes[3] << 32)
+            | ((unsigned long long)oid_bytes[4] << 24)
+            | ((unsigned long long)oid_bytes[5] << 16)
+            | ((unsigned long long)oid_bytes[6] << 8)
+            | ((unsigned long long)oid_bytes[7]);
+        oid_str = PyUnicode_FromFormat(" oid 0x%x", oid_value);
```
Comment (mgedmin) [inline persistent/cPersistence.c]: [The documentation](https://docs.python.org/3/c-api/unicode.html) says %x takes an `int`, not an `unsigned long long`.

There are many platforms where an `int` is shorter than 64 bits.

We should at least have a unit test that uses an OID that doesn't fit in 32 bits.

File: persistent/cPersistence.c [hunk 2]
```diff
@@ -1446,23 +1451,50 @@ Per_repr(cPersistentObject *self)
...
+    if (self->oid && PyBytes_Check(self->oid) && PyBytes_GET_SIZE(self->oid) == 8) {
+        oid_bytes = (unsigned char*)PyBytes_AS_STRING(self->oid);
+        oid_value = ((unsigned long long)oid_bytes[0] << 56)
+            | ((unsigned long long)oid_bytes[1] << 48)
+            | ((unsigned long long)oid_bytes[2] << 40)
+            | ((unsigned long long)oid_bytes[3] << 32)
+            | ((unsigned long long)oid_bytes[4] << 24)
+            | ((unsigned long long)oid_bytes[5] << 16)
+            | ((unsigned long long)oid_bytes[6] << 8)
+            | ((unsigned long long)oid_bytes[7]);
```
Comment (mgedmin) [inline persistent/cPersistence.c]: Could we use uint64_t instead of unsigned long long?

File: persistent/tests/test_persistence.py [hunk 3]
```diff
@@ -1733,12 +1733,27 @@ def __repr__(self):
...
         result = self._normalized_repr(p)
         self.assertEqual(
             result,
-            "<persistent.Persistent object at 0xdeadbeef oid b'12345678'>")
+            "<persistent.Persistent object at 0xdeadbeef oid " + self._HEX_OID + ">")
+
+    def test_64bit_oid(self):
+        import struct
+        p = self._makeOne()
+        oid_value = 2 << 62
```
Comment (mgedmin) [inline persistent/tests/test_persistence.py]: 1 << 63 would be clearer imho

**Comment by jamadden** [comment]
Thanks! I don’t consider myself a c api expert either, so I welcome all reviews.

I know that downstream is waiting on this though, and because I feel bad that my implementation of the requested improvements broke things, so if there are no further comments in, say, the next 36 hours, I expect to go ahead and merge and release.

File: persistent/tests/test_persistence.py (hunk 3 above)
Comment (mgedmin) [reply persistent/tests/test_persistence.py]: I was looking at this, wondering why 2**62 instead of 2**63, and then I realized << is not ** :D

**Comment by icemac** [comment]
@jamadden +1

**Comment by icemac** [comment]
@jamadden 👍

**Comment by Rotonen** [comment]
@jamadden I owe you a `$BEVERAGE` should our paths ever cross.

**Comment by jamadden** [comment]
4.4.3 is on PyPI.

**Pull request merged**